packages/plugin-core/src/commands/ConvertLink.ts

```diff
@@ -16,31 +44,310 @@ export class ConvertLinkCommand extends BasicCommand<
 > {
   key = DENDRON_COMMANDS.CONVERT_LINK.key;
 
-  async gatherInputs(_opts: CommandOpts): Promise<CommandOpts> {
-    return _opts;
+  noAvailableOperationError = new DendronError({
+    message: `No available convert operation for link at cursor position.`,
+  });
+
+  noVaultError = new DendronError({
+    message: "this link points to a note in a vault that doesn't exist",
+  });
+
+  noLinkError = new DendronError({
+    message: `No link at cursor position.`,
+  });
+
+  noTextError = new DendronError({
+    message: "Failed to determine text to replace broken link.",
+  });
+
+  prepareBrokenLinkConvertOptions(reference: getReferenceAtPositionResp) {
+    const parsedLink = LinkUtils.parseLinkV2({
+      linkString: reference.refText,
+      explicitAlias: true,
+    }) as ParseLinkV2Resp;
+    const aliasOption: QuickPickItem = {
+      label: "Alias",
+      description: parsedLink?.alias,
+      detail: "Convert broken link to alias text.",
+    };
+    const hierarchyOption: QuickPickItem = {
+      label: "Hierarchy",
+      description:
+        reference.refType === "usertag" || reference.refType === "hashtag"
+          ? reference.ref
+          : parsedLink?.value,
+      detail: "Convert broken link to hierarchy.",
+    };
+    const noteNameOption: QuickPickItem = {
+      label: "Note name",
```

Comment:
it's already a broken link and we don't have access to the note prop that existed before. that's why I opted for using the note name

Reply:
maybe we could enhance it later to use the note title when we decide to chain this with `delete node`. For that case we can get the note title before deleting.

I also added a task to add a command to support general manipulation of existing valid links so that could be another place we can support using note titles.